gr-blocks/include/gnuradio/blocks/file_meta_sink.h

```diff
@@ -88,15 +88,15 @@ namespace gr {
        * \param max_segment_size (size_t): Length of a single segment
        *    before the header is repeated (in items).
        * \param extra_dict (string): a serialized PMT dictionary of extra
-       *    information. Currently not supported.
+       *    information.
        * \param detached_header (bool): Set to true to store the header
        *    info in a separate file (named filename.hdr)
        */
       static sptr make(size_t itemsize, const std::string &filename,
 		       double samp_rate=1, double relative_rate=1,
 		       gr_file_types type=GR_FILE_FLOAT, bool complex=true,
 		       size_t max_segment_size=1000000,
-		       const std::string &extra_dict="",
+           const std::vector<uint8_t> &extra_dict=std::vector<uint8_t>(),
```

Comment:
This is an API change (which is OK, we're on master). However, when do we express serialized `pmt_t` as `std::vector<uint8_t>` in C++-land? I thought it was always `string`.

Reply:
When I tried directly exporting `std::vector<uint8_t>`, it converted it to tuples in Python, which is not what we wanted. My wrappers goes to std::vector<uint8_t>` intermediately because I need to avoid the UTF8 conversion, then my custom Python will turn that into a bytes string.

Reply:
Exactly. That's the logical step to take – we don't want the output of pmt.serialize to be interpreted as unicode – it's just an array of 8 bit numbers.

Now, this change makes the complementary operation – feeding a serialized PMT back into C++-land – possible. It happens that `vector<uint8_t>` is something that SWIG is able to convert `bytes` to. That's why it was chosen (I discovered that you were converting through `vector<uint8_t>` only after figuring out how to accept `bytes` from pythonland in c++land).

Reply:
So, between a rock and a hard place:

* C++ `pmt::serialize` gives us `std::string`
* We can't accept `std::string` here, because we can't teach SWIG that it can convert `bytes` to `std::string`.

Reply:
Actually, yeah, I'm not sure why the block wasn't already doing that :)

Have I recently mentioned my feelings for SWIG? :grimacing: